lib/matplotlib/table.py

```diff
@@ -453,19 +476,28 @@ def _auto_set_font_size(self, renderer):
 
         if len(self._cells) == 0:
             return
-        fontsize = list(six.itervalues(self._cells))[0].get_fontsize()
-        cells = []
+
+        fontsize = None
```

Comment:
except that would not work.

I am using fontsize == None to spot first time round the loop, when the font can grow or shrink.